pkg/reconciler/dependencybuild/buildrecipeyaml.go

```diff
@@ -55,11 +53,9 @@ var hermeticBuildEntryScript string
 
 func createPipelineSpec(tool string, commitTime int64, jbsConfig *v1alpha12.JBSConfig, systemConfig *v1alpha12.SystemConfig, recipe *v1alpha12.BuildRecipe, db *v1alpha12.DependencyBuild, paramValues []pipelinev1beta1.Param, buildRequestProcessorImage string) (*pipelinev1beta1.PipelineSpec, string, error) {
 
-	marshaled, err := json.Marshal(recipe)
-	if err != nil {
-		return nil, "", err
-	}
-	imageId := util.HashString(string(marshaled) + buildRequestProcessorImage + tool + db.Name)
+	// Rather than tagging with hash of json build recipe, buildrequestprocessor image and db.Name as the former two
```

Comment:
I don't think we want to change this imageId here. I think we just want a new additional tag with the name.

I don't even know if the current imageId was the correct choice, or if it should have just been random. 

Basically my thinking is that if you ever need to rebuild something because of a problem you might still want to reference the old artifacts, but because we are not renaming currently this will just overwrite all the tags. If we have a random image id then there will always be a reference to the old image so the legacy artifacts will still be accessible (kinda, assuming you have stored this id somewhere). 

That said as this is not really implemented properly anywhere yet making the imageId the name and then creating the concept of an archiveId or something at a later date is probably the way to go.